src/main/java/nl/hr/recipefinder/controller/RecipeController.java

```diff
@@ -28,37 +34,48 @@ public RecipeController(RecipeService recipeService, ModelMapper modelMapper) {
     this.modelMapper = modelMapper;
   }
 
-
   @GetMapping()
-  public List<ListedRecipeDto> getRecipes() {
+  public ResponseEntity<List<ListedRecipeDto>> getRecipes() {
     List<Recipe> recipes = recipeService.getRecipes();
-
-    return recipes.stream()
-      .map((it) -> modelMapper.map(it, ListedRecipeDto.class))
-      .collect(Collectors.toList());
+    try {
+      return new ResponseEntity<>(recipes.stream()
+        .map((it) -> modelMapper.map(it, ListedRecipeDto.class))
+        .collect(Collectors.toList()), HttpStatus.OK);
+    } catch (Exception e) {
+      throw new HttpInternalServerError();
+    }
   }
 
-
   @GetMapping("/search/{searchInput}")
-  public List<ListedRecipeDto> searchRecipes(@PathVariable String searchInput) {
-    List<Recipe> recipes = recipeService.findRecipesByNameAndDescription(searchInput, searchInput);
+  public ResponseEntity<List<ListedRecipeDto>> searchRecipes(@PathVariable String searchInput) {
+    try {
+      List<Recipe> recipes = recipeService.findRecipesByNameAndDescription(searchInput, searchInput);
 
-    return recipes.stream()
-      .map((it) -> modelMapper.map(it, ListedRecipeDto.class))
-      .collect(Collectors.toList());
+      return new ResponseEntity<>(recipes.stream()
+        .map((it) -> modelMapper.map(it, ListedRecipeDto.class))
+        .collect(Collectors.toList()), HttpStatus.OK);
+    } catch (Exception e) {
+      throw new HttpInternalServerError();
+    }
   }
 
   @GetMapping("/{id}")
-  public RecipeDto Recipe(@PathVariable("id") Long id) {
+  public ResponseEntity<RecipeDto> Recipe(@PathVariable("id") Long id) {
     Optional<Recipe> recipe = recipeService.findById(id);
-    if (recipe.isPresent()) return modelMapper.map(recipe.get(), RecipeDto.class);
+    if (recipe.isPresent()) return new ResponseEntity<>(modelMapper.map(recipe.get(), RecipeDto.class), HttpStatus.OK);
```

Comment:
Dit is eigenlijk niet nodig omdat Spring automatisch er een `ResponseEntity` van maakt, maar het maakt het wel wat meer verbose dus ik heb er niks op tegen

Reply:
Voornaamste voordeel door expliciet ResponseEntity te returnen is meer controle over welke statuscode de backend teruggeeft en wanneer dit gebeurt. Dit voegt waarde toe zodra een methode 201, 4XX's, 5XX's e.d. terug dienen te geven.